compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2098,6 +2120,7 @@ impl<'tcx> Ty<'tcx> {
         T::collect_and_apply(iter, |ts| Ty::new_tup(tcx, ts))
     }
 
+    /// Creates a new [`Ty`] representing a function definition type, with `def_id` and `args` generic args.
```

Comment:
```suggestion
    /// Create a new [`Ty`] representing a function definition type.
```

`DefId` and `[GenericArg<'_>]` are so common in the codebase that if we mention them here we'd also have to document them in almost every place, where their type docstring should have enough explanation on what they do, so I'd rather not even mention them here.